sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/ForbidCertainMethodCheck.java

```diff
@@ -0,0 +1,267 @@
+import java.util.regex.Pattern;
+
+import com.github.sevntu.checkstyle.Utils;
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.CheckstyleException;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.utils.CommonUtils;
+
+/**
+ * Check that forbidden method is not used. We can forbid a method by method name and number of
+ * arguments.
+ * This can be used to enforce things like:
+ * <ul>
+ * <li> System.exit() should not be called.</li>
+ * <li> Assert.assertTrue() and Assert.assertFalse() have a 1 arg variant that does not provide a
+ * helpful message on failure.
+ * </ul>
+ * Parameters are:
+ * <ul>
+ * <li><b>methodName</b> - Regex to match name of the method to be forbidden.
+ * When blank or unspecified, all the methods are allowed.</li>
+ * <li><b>argumentCount</b> - Number or range to match number of arguments the method takes.
+ * Multiple numbers/ranges must be comma separated. When unspecified, defaults to "0-".
+ * </ul>
+ *
+ * <p>An example configuration:
+ * <pre>
+ * &lt;module name="ForbidCertainMethodCheck"&gt;
+ *    &lt;property name="methodName" value="exit"/&gt;
+ * &lt;/module&gt;
+ * &lt;module name="ForbidCertainMethodCheck"&gt;
+ *    &lt;property name="methodName" value="assert(True|False)"/&gt;
+ *    &lt;property name="argumentCount" value="1"/&gt;
+ * &lt;/module&gt;
+ * &lt;module name="ForbidCertainMethodCheck"&gt;
+ *    &lt;property name="methodName" value="assertEquals"/&gt;
+ *    &lt;property name="argumentCount" value="2"/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * Argument count can be bounded range (e.g.: {@code 2-4}) or unbounded range
+ * (e.g.: {@code -5, 6-}). Unbounded range can be unbounded only on one side.
+ * Multiple ranges must be comma separated.
+ * For example, the following will allow only 4 and 8 arguments.
+ *
+ * <pre>
+ * &lt;module name="ForbidCertainMethodCheck"&gt;
+ *    &lt;property name="methodName" value="asList"/&gt;
+ *    &lt;property name="argumentCount" value="-3, 5-7, 9-"/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ *
+ * @author <a href="mailto:raghavgautam@gmail.com">Raghav Kumar Gautam</a>
+ */
+public class ForbidCertainMethodCheck extends AbstractCheck {
+
+    /** Key is pointing to the warning message text in "messages.properties" file. */
+    public static final String MSG_KEY = "forbid.certain.method";
+    /** Regex for splitting string on comma. */
+    private static final Pattern COMMA_REGEX = Pattern.compile(",");
+
+    /** Name of the method. */
+    private Pattern methodName = CommonUtils.createPattern("^$");
+
+    /** Range for number of arguments. */
+    private String argumentCount = "0-";
+    /** Range objects for matching number of arguments. */
+    private final List<IntRange> argumentCountRanges = new ArrayList<>(
+        Collections.singletonList(new IntRange(0, null)));
```

Comment:
limits or range are by Integer type, so please remove usage of `null` and use Integer.MAX.
IntRange should fail if null is passed to it.

Reply:
Done. Also changed Integer to int.